File: web/src/api/file.ts
import request, { type PageResult, type PageParams } from './request'

export interface FileRecord {
  id: number
  file_name: string
  original_name: string
  file_key: string
  file_size: number
  mime_type: string
  storage_driver: string
  url: string
  user_id: number
  created_at: string
  deleted_at?: string
}

export interface FileListParams extends PageParams {
  mime_type?: string
}

export interface UploadResult {
  id: number
  file_name: string
  original_name: string
  file_key: string
  file_size: number
  mime_type: string
  url: string
}

const fileApi = {
  upload(file: File) {
    const formData = new FormData()
    formData.append('file', file)
    return request.post<any, { code: number; data: UploadResult }>('/files/upload', formData, {
      headers: { 'Content-Type': 'multipart/form-data' }
    })
  },

  list(params?: FileListParams) {
    return request.get<any, { code: number; data: PageResult<FileRecord> }>('/files/list', { params })
  },

  download(id: number) {
    return request.get(`/files/${id}`, { responseType: 'blob' })
  },

  delete(id: number) {
    return request.delete(`/files/${id}`)
  }
}

export default fileApi

File: backend/internal/model/file.go
package model

type FileRecord struct {
	SnowflakeTenantModel
	FileName      string `gorm:"type:varchar(256);not null" json:"file_name"`
	OriginalName  string `gorm:"type:varchar(256);not null" json:"original_name"`
	FileKey       string `gorm:"type:varchar(512);not null;index" json:"file_key"`
	FileSize      int64  `gorm:"type:bigint;not null" json:"file_size"`
	MimeType      string `gorm:"type:varchar(128)" json:"mime_type"`
	StorageDriver string `gorm:"type:varchar(20);default:'local'" json:"storage_driver"`
	URL           string `gorm:"type:varchar(1024)" json:"url"`
	UserID        int64  `gorm:"index" json:"user_id"`
}

func (FileRecord) TableName() string { return "file_records" }
